content/blog/2020-11-25-november-20-community-gems.md

```diff
@@ -0,0 +1,195 @@
+---
+title: November '20 Community Gems
+date: 2020-11-25
+description: |
+  A roundup of technical Q&A's from the DVC community. This month, learn how
+  to clean your cache and use Git hooks with DVC. And here's an early holiday 
+  gift- new Bitbucket support for CML!
+descriptionLong: |
+  A roundup of technical Q&A's from the DVC community. This month, learn how
+  to clean your cache and use Git hooks with DVC. And here's an early holiday 
+  gift- new Bitbucket support for CML!
+picture: 2020-11-25/cover.png
+author: elle_obrien
+commentsUrl: https://discuss.dvc.org/t/october-20-community-gems/535
+tags:
+  - Discord
+  - Gems
+  - CML
+  - Cache
+  - Bitbucket
+---
+
+## DVC questions
+
+### [Q: If I make a new Git branch, how can I synchronize it with DVC?](https://discord.com/channels/485586884165107732/485596304961962003/773498570795778058)
+
+Generally, yes. When you make a new Git branch in your project:
+
+```dvc
+$ git checkout -b <my_great_new_branch>
+```
+
+you'll want to next run
+
+```dvc
+$ dvc checkout
+```
+
+to synchronize your `.dvc` files on that branch. But _did you know_ you can
+automate this with a `post-checkout` Git hook? We've got a hook that executes
+`dvc checkout` whenever you run `git checkout`, so you'll always have the
+correct data file versions. Head to our docs to
+[read up on installing Git hooks into your DVC repository](https://dvc.org/doc/command-reference/install#install)
+so you never forget to `dvc checkout`!
+
+### [Q: I have a big, 100 GB directory. I want to know where the contents are located so I can open them with Spark- is there a way to get the location of my files without caching them locally?](https://discord.com/channels/485586884165107732/485596304961962003/771386223403073587)
+
+For this, we'd recommend the DVC Python API's `get_url` function. For example,
+in a Python script you'd write:
+
+```python
+import dvc.api
+
+resource_url = dvc.api.get_url(
+  "<top-level-directory>",
+  repo="https://github.com/<your-repo>")
+)
+```
+
+This code means the API will return the URL for a file that ends in `.dir`. The
+`.dir` file contains a JSON-formatted table of the hashes and relative paths for
+all the files inside `<top-level-directory>`. You could then parse that file to
+tget the locations of the full directory contents.
+
+Why use the API? It's a good way to access DVC-tracked files without downloading
```

Comment:
so, in this case it doesn't provide much of a help - you have to parse JSON, and you are pretty much on your own- have to use specific cloud library to stream/downloads.

Reply:
i agree that you get very little help for parsing the data, but I think there's something important about using the API to avoid downloading the huge directory.

Do you think this person's question could be better answered via `dvc list <myremote>`, and then appending the paths of those files to the remote path in your `.dvc/config`?

Reply:
> but I think there's something important about using the API

it is, but in this case it's not DVC API that will be used (DVC actually even makes it harder - you need to collect file names in the cloud to access this data).

the answer is fine, the only part about `Why use the API?` to my mind is misleading a bit. In this case the answer to that question is that we have to use it to get the list of file since DVC stores files in the content-addressable way, not because it gives us a way to access something w/o downloading.

Better probably give an example here, like Peter did later in the discussion how does path is created